presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -1457,6 +1462,93 @@ protected Node aggregateResult(Node aggregate, Node nextResult)
         throw new UnsupportedOperationException("not yet implemented");
     }
 
+    private static String decodeString(SqlBaseParser.UnicodeStringContext context)
+    {
+        final char delimiter;
+        if (context.UESCAPE() != null) {
+            String delimiterString = context.BASIC_STRING().getText();
+            if (delimiterString.length() != 3) {
+                throw parseError("Invalid escaped delimiter: " + delimiterString, context);
+            }
+            delimiter = unquote(delimiterString).charAt(0);
+            if (isHexDigit(delimiter) || delimiter == '"' || delimiter == '+' || delimiter == ' ') {
+                throw parseError("Invalid escaped delimiter: " + delimiterString, context);
+            }
+        }
+        else {
+            delimiter = '\\';
+        }
+
+        String rawContent = unquote(context.UNICODE_STRING().getText().substring(2));
+        StringBuilder builder = new StringBuilder();
+        int beginCodeIndex = 0;
+        int charactersNeeded = 0;
+        for (int i = 0; i < rawContent.length(); i++) {
+            if (rawContent.charAt(i) == delimiter) {
+                if (charactersNeeded != 0) {
+                    throw parseError("Incomplete escaped character: " + rawContent.substring(beginCodeIndex, i + 1), context);
+                }
+                if (i < rawContent.length() - 1) {
```

Comment:
Invert this `if` block to throw early, then the `else` block can be removed and made unconditional (removes one level of indentation)